iot_packet_verifier/src/balances.rs

```diff
@@ -61,41 +54,52 @@ impl BalanceCache {
         }
 
         Ok(Self {
-            sub_dao,
-            provider,
             balances: Arc::new(Mutex::new(balances)),
+            solana,
         })
     }
+}
 
+impl<S> BalanceCache<S> {
```

Comment:
shouldn't this `balances` method be grouped with the previous `impl<S> BalanceCache<S>` above?

Reply:
It doesn't depend on S implementing `Debiter`